src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/DotnetCliHelper.cs

```diff
@@ -68,14 +68,15 @@ private async Task<string> GetDotnetSdkFolderFromDotnetExecutableAsync(string pr
         return dotnetSdkFolderPath;
     }
 
-    public Process Run(string[] arguments, string? workingDirectory, bool shouldLocalizeOutput)
+    public Process Run(string[] arguments, string? workingDirectory, bool shouldLocalizeOutput, bool redirectStandardInput = false)
     {
         _logger.LogDebug($"Running dotnet CLI command at {_dotnetExecutablePath.Value} in directory {workingDirectory} with arguments {arguments}");
 
         var startInfo = new ProcessStartInfo(_dotnetExecutablePath.Value)
         {
             CreateNoWindow = true,
             UseShellExecute = false,
+            RedirectStandardInput = redirectStandardInput,
```

Comment:
Should this frankly always be true as well? Because it seems either that:

1. The command we're running won't read stdin (so it won't really matter).
2. The command we might be running might incidentally read stdin (like restore triggering an interactive credential provider) and we probably don't want it using some random stdin from...somewhere?

Reply:
In case (2), I think it would essentially read the stdin of the parent process, which it seems like we wouldn't want.